usaspending_api/common/elasticsearch/json_helpers.py

```diff
@@ -12,5 +13,12 @@ def json_str_to_dict(string: str) -> dict:
 
     try:
         return json.loads(string.encode("unicode_escape"))
+    except json.decoder.JSONDecodeError:
+        pass  # One more attempt to Decode
+
+    try:
+        regex = r"\"([^\"]*)\"\s?:\s?\"([^\"]*(?:(?:\w|\s)?(?:\"|\')?(?:\w|\s)?)*[^\"]*)(?:(?:\"\,)|(?:\"\}))"
+        matches = re.findall(regex, string)
+        return {key: value for key, value in matches}
     except json.decoder.JSONDecodeError as e:
```

Comment:
Does this exception make sense here? What checks should be done to ensure the regex pickups everything intended?

Reply:
Good point. Made a change to ensure that the number of results match what we expect and also throw the DecodeError if we couldn't salvage with the Regex.